uni/lib/view/Widgets/schedule_row.dart

```diff
@@ -45,13 +52,34 @@ class ScheduleRow extends StatelessWidget {
                             mainAxisAlignment: MainAxisAlignment.center,
                             crossAxisAlignment: CrossAxisAlignment.center,
                             children: [
-                              ScheduleTimeInterval(begin: begin, end: end)
+                              ScheduleTimeInterval(
+                                  begin: widget.exam.begin,
+                                  end: widget.exam.end)
                             ]),
-                        ScheduleEventRectangle(subject: subject, type: type),
-                        IconButton(
-                            icon: const Icon(MdiIcons.calendarPlus, size: 30),
-                            onPressed: () =>
-                                Add2Calendar.addEvent2Cal(createExamEvent())),
+                        ScheduleEventRectangle(
+                            subject: widget.exam.subject,
+                            type: widget.exam.examType),
+                        Column(children: [
+                          IconButton(
+                              icon: const Icon(MdiIcons.calendarPlus, size: 30),
+                              onPressed: () =>
+                                  Add2Calendar.addEvent2Cal(createExamEvent())),
+                          if (!widget.mainPage)
+                            IconButton(
+                                icon: widget.isHidden
+                                    ? const Icon(Icons.remove_red_eye_outlined,
+                                        size: 30)
+                                    : const Icon(Icons.remove_red_eye,
+                                        size: 30),
```

Comment:
We can use a differing icon to hidden the exam and not changing it style (outlining to filling)
Consider using `Icons.visibility` and `Icons.visibility_off` like the login page.
Or explore the MdiIcons package as we already are importing here.
Some interesting ones: `eye-off` or `eye-remove`